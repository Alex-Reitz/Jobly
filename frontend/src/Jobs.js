import React from "react";

function Jobs() {
  return (
    <div>
      <h1>This is the Jobs component</h1>
    </div>
  );
}

export default Jobs;
