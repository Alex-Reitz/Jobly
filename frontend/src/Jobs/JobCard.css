.job-content {
  text-align: left;
  margin: 20px;
  padding: 5px;
  outline: 2px solid black;
  width: 40%;
  display: inline-block;
}
.jobcard-company-name {
  margin: 3px;
}
.job-title-card {
  margin: 3px;
}
.job-salary-card {
  margin: 3px;
}
.job-equity-card {
  margin: 3px;
}
.job-application-card {
  margin: 3px;
}
.job-card-apply {
  background-color: #316685;
  border: none;
  color: white;
  padding: 10px;
  text-decoration: none;
  margin: 4px 2px;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 5px;
  float: right;
}
.job-card-applied {
  display: none;
}
.job-status {
  text-decoration: none;
  margin: 4px 2px;
  font-size: 1rem;
  border-radius: 5px;
  float: right;
}
