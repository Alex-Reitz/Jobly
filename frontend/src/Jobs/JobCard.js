import React, { useContext } from "react";
import JoblyApi from "../api";
import UserContext from "../Context/userContext";

/* {id: 20, title: "Tourist information centre manager", salary: 88000, equity: "0", companyHandle: "foster-rice", â€¦}
companyHandle: "foster-rice"
companyName: "Foster-Rice"
equity: "0"
id: 20
salary: 88000
title: "Tourist information centre manager" */

function JobCard({ info }) {
  const user = useContext(UserContext);
  const apply = (username, jobid) => {
    async function applyToJob(username, jobid) {
      const res = await JoblyApi.apply(user.username, info.id);
      console.log(res);
      return res;
    }
    applyToJob(username, jobid);
  };
  return (
    <div className="job-content">
      <div className="company-title">
        {info.companyName ? (
          <p>
            <strong>Company Name: </strong>
            {info.companyName}
          </p>
        ) : undefined}
      </div>
      <div className="job-title">
        {info.title ? (
          <p>
            <strong>Job Title: </strong>
            {info.title}
          </p>
        ) : undefined}
      </div>
      <div className="job-salary">
        {info.salary ? (
          <p>Job Salary: {info.salary}</p>
        ) : (
          <p>Sorry, no salary data to show.</p>
        )}
      </div>
      <div className="job-equity">
        {info.equity > 0 ? (
          <p>Job equity: {info.equity}</p>
        ) : (
          <p>Sorry, no equity data to show.</p>
        )}
      </div>
      <button onClick={apply}>Apply</button>
      <hr></hr>
    </div>
  );

  /* if (!info.equity && !info.salary) {
    return (
      <div className="content">
        <p>
          <strong>Company Name: </strong>
          {info.companyName}
        </p>
        <p>
          <strong>Job Title: </strong>
          {info.title}
        </p>
        <button>Apply</button>
        <hr />
      </div>
    );
  } else if ((!info.equity || info.equity === "0") && info.salary) {
    return (
      <div className="content">
        <p>
          <strong>Company Name: </strong>
          {info.companyName}
        </p>
        <p>
          <strong>Job Title: </strong>
          {info.title}
        </p>
        <p>
          <strong>Salary:</strong>${info.salary}
        </p>
        <p>Equity Options Not Available</p>
        <button>Apply</button>
        <hr></hr>
      </div>
    );
  } else if (!info.salary) {
    return (
      <div className="content">
        <p>
          <strong>Company Name: </strong>
          {info.companyName}
        </p>
        <p>
          <strong>Job Title: </strong>
          {info.title}
        </p>
        <p>Salary Options Not Available</p>
        <p>
          <strong>Equity:</strong>
          {info.equity}
        </p>
        <button>Apply</button>
        <hr></hr>
      </div>
    );
  } else {
    return (
      <div className="content">
        <p>
          <strong>Company Name: </strong>
          {info.companyName}
        </p>
        <p>
          <strong>Job Title: </strong>
          {info.title}
        </p>
        <p>
          <strong>Salary:</strong>${info.salary}
        </p>
        <p>
          <strong>Equity:</strong>
          {info.equity}
        </p>
        <button>Apply</button>
        <hr></hr>
      </div>
    );
  } */
}

export default JobCard;
