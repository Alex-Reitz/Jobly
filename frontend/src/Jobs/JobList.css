.logged-in-warning {
  color: black;
  text-align: center;
  font-size: 2rem;
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-transform: capitalize;
}
.job-heading {
  color: black;
  text-align: center;
  font-size: 2rem;
  margin: 10px;
}
.search {
  margin: 10px;
}
.search-jobs-label {
  color: black;
  font-size: 1rem;
  padding: 3px;
}
.job-title-search-input {
  width: 15%;
  padding: 0.25em;
  margin: 5px;
}
