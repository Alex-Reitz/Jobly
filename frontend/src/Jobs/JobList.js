import React, { useState, useEffect, useContext } from "react";
import JobCard from "./JobCard";
import JoblyApi from "../api";
import UserContext from "../Context/userContext";
import "./JobList.css";

function JobList() {
  const [jobFormData, setJobFormData] = useState({
    title: "",
  });
  const [jobs, setJobs] = useState([]);
  const { currentUser } = useContext(UserContext);

  //Gets jobs list from backend on page load
  useEffect(() => {
    async function getJobs() {
      const res = await JoblyApi.getJobs();
      setJobs(res.jobs);
      return res.jobs;
    }
    getJobs();
  }, []);

  //function to get all jobs by title matching a query from the search form, returns all jobs if form is blank on submission
  const searchJobNames = (data) => {
    if (data.title.length === 0) {
      async function getJobs() {
        const res = await JoblyApi.getJobs();
        setJobs(res.jobs);
        return res.jobs;
      }
      getJobs();
    } else {
      async function searchName() {
        let res = await JoblyApi.searchJobs(data);
        setJobs(res.jobs);
        return res.jobs;
      }
      searchName();
    }
  };

  //Handle form change
  const handleJobChange = (evt) => {
    const { name, value } = evt.target;
    setJobFormData((jobFormData) => ({
      ...jobFormData,
      [name]: value,
    }));
  };
  //Gther input from form on submission
  const gatherJobInput = (evt) => {
    evt.preventDefault();
    searchJobNames({ ...jobFormData });
  };
  if (!currentUser.username) {
    return (
      <div>
        <h1 className="logged-in-warning">
          You must be logged in to access the jobs list
        </h1>
      </div>
    );
  } else {
    return (
      <div className="joblist-container">
        <h1 className="job-heading">Take a look at these Jobs!</h1>
        <div className="search">
          <form onSubmit={gatherJobInput}>
            <label className="search-jobs-label" htmlFor="title">
              Search Job Titles
            </label>
            <input
              className="job-title-search-input"
              onChange={handleJobChange}
              type="text"
              name="title"
              value={jobFormData.title}
              id="title"
            />
          </form>
          <hr></hr>
        </div>
        {jobs.map((job) => (
          <JobCard key={job.id} info={job} />
        ))}
      </div>
    );
  }
}

export default JobList;
