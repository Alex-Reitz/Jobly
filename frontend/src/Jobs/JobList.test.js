import React from "react";
import { render } from "@testing-library/react";
import Jobs from "./JobList";
import { MemoryRouter } from "react-router";
import UserContext from "../Context/userContext";

it("matches snapshot", function () {
  const testFunc = jest.fn();
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: testFunc }}>
        <Jobs />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
