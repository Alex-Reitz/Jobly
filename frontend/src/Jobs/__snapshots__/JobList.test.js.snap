// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="logged-in-warning"
    >
      You must be logged in to access the jobs list
    </h1>
  </div>
</DocumentFragment>
`;
